docs/built_in_extensions.rst

```diff
@@ -108,6 +108,37 @@ The HTTP entrypoint is built on top of `werkzeug <http://werkzeug.pocoo.org/>`_.
     Date: Fri, 13 Feb 2015 14:58:48 GMT
 
 
+Exceptions can be raised safely if added to ``expected_exceptions``. You can also change exceptions formatting by overriding :meth:`~nameko.web.HttpRequestHandler.response_from_exception`:
```

Comment:
@davidszotten I think that's probably the case. I intended for it to be a public API on all entrypoints, but we only had experience of using it in RPC. This PR obviously solves some of that.

The docs need a bit of a revamp. The "built-in extensions" content is pretty light. I always intended  to fill in the detail on the underlying implementations, and I think we also need somewhere to document these features that are common to many implementations (``expected_exceptions``, ``sensitive_variables``, the whole concept of communication between extensions etc.)